adiachenko/src/components/FilmsList/index.tsx

```diff
@@ -25,19 +26,34 @@ const FoundCountStyled = styled.div`
 const FilmsList = () => {
 
   const [films, setFilms] = useState([]);
+  const [sortedFilms, setSortedFilms] = useState([]);
   const [isLoading, setLoading] = useState(false);
+  const contextValue = useAppContext();
+
+  const sortingByCategory = useCallback((filmsList: FilmProps[], type: string) => {
+    if (type === 'All') {
+      return filmsList;
+    }
+
+    return filmsList.filter(film => film.genres.includes(type))
+  }, []);
 
   useEffect(() => {
     setLoading(true);
 
     setTimeout(() => {
```

Comment:
I believe, this one is here to wait till something happens (data fetch or whatever it might be). If so, it is a bad practice. Bear in mind, you may pass `async` function to `useEffect()` and use `await` to maintain the desired execution order. Otherwise, you introduce intentional 5s lag (even when it might not be needed), still taking a risk that if 5s is not enough (abnormal network connection delays, etc) you run in the very same issue, you're trying to solve.

Reply:
This part was added to emulate data loading for this phase and will be removed in the next phase.